src/dalle_mini/model/modeling.py

```diff
@@ -561,3 +563,24 @@ def _decoder_forward(
             outputs = outputs[:1] + (unfreeze(past["cache"]),) + outputs[1:]
 
         return outputs
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
+        """
+        Initializes from a wandb artifact, or delegates loading to the superclass.
+        """
+        if ":" in pretrained_model_name_or_path and not os.path.isdir(
+            pretrained_model_name_or_path
+        ):
+            # wandb artifact
+            artifact = wandb.Api().artifact(pretrained_model_name_or_path)
+
+            # we download everything, including opt_state, so we can resume training if needed
+            # see also: #120
+            pretrained_model_name_or_path = artifact.download()
+
+        model = super(DalleBart, cls).from_pretrained(
+            pretrained_model_name_or_path, *model_args, **kwargs
+        )
+        model.config.resolved_name_or_path = pretrained_model_name_or_path
```

Comment:
The issue here is that it will also be saved in the config file, which is not ideal…
Maybe we could just create a `DalleTokenizer` in a follow-up PR that does the same? In theory it should not be redownloaded because it will be cached.

Reply:
Yes, we can do that or save the property inside the model instead of the config. (Both model and config classes [already have a `name_or_path` property](https://github.com/huggingface/transformers/blob/v4.15.0/src/transformers/modeling_utils.py#L482), but I'm not sure when it is populated).

One reason for doing it this way (instead of using an overridden Tokenizer) is that we can use the same property to load the optimizer state, as I did as part of this PR. Of course, if we go the route of #120, then that would not be necessary.

What do you think?

Reply:
Thinking about it, I guess that overriding the Tokenizer would be more consistent with the HF API. We could do the same for the configuration, just to make sure we have all the pieces.

Reply:
Sounds like a plan!